src/main/java/dk/cngroup/lentils/service/HintService.java

```diff
@@ -43,6 +49,19 @@ public Hint getHint(final Long hintId) {
         return hintRepository.saveAll(hints);
     }
 
+    public void addImageToHintAndSave(final HintFormDTO hintFormDto,
+                                      final Hint hint,
+                                      final Image oldImage) {
+        if (!FileTreatingUtils.isFilePresentInHintForm(hintFormDto)) {
+            hint.setImage(oldImage);
+            save(hint);
```

Comment:
`save` hint bych vytahl z if a else branche a dal bych ho az za celou if/else konstrukci, aby byl volan pouze jednou.

Reply:
To je dobrý nápad.
Možná by šlo ty bloky dát do samostatných privátních metod v té podmínce. Ať tam těch ifů pak nevzniká víc tady. No uvidíš.

Reply:
přesunula jsem save(Hint), a ještě jsem musela upravit deleteImage kvůli NullPointerException